numba/cuda/cudadrv/devicearray.py

```diff
@@ -354,15 +354,37 @@ def view(self, dtype):
         copy of the data.
         """
         dtype = np.dtype(dtype)
-        if dtype.itemsize != self.dtype.itemsize:
-            raise TypeError("new dtype itemsize doesn't match")
+        shape = list(self.shape)
+        strides = list(self.strides)
+
+        if self.dtype.itemsize != dtype.itemsize:
+            if not self.is_c_contiguous():
+                raise ValueError(
+                    "To change to a dtype of a different size,"
+                    " the array must be C-contiguous"
+                )
+
+            shape[-1], rem = divmod(
+                shape[-1] * self.dtype.itemsize,
+                dtype.itemsize
+            )
```

Comment:
Perhaps don't mutate the `shape` until the `rem != 0` check is completed, else someone could do this `.view()` call in a `try/except ValueError` and end up with a mutated shape surprise?

Reply:
Ignore this, the shape is already a copy! Missed that!

Reply:
No worries 🙂